src/aggregator.js

```diff
@@ -8,72 +8,103 @@
  *  LICENSE file in the root directory of this source tree.
  */
 
+import _ from "underscore";
 import Generator from "./generator";
 
+/**
+ * An aggregator takes the following options:
+ *
+ *     - 'window'    -     size of the window to aggregate over (e.g. "5m")
+ *     - 'operator'  -     function (e.g. avg)
+  *    - 'fieldSpec' -     (optional) describes what part of the events to
+ *                         include in the aggregation. Maybe a string, list
+ *                         of strings for each event column, or a function.
+ *                         If a function it should return a list of key/values
+ *                         in an object.
+ *     - 'emit'      - (optional) Rate to emit events. Either:
+ *                         "always" - emit an event on every change
+ *                         "next" - just when we advance to the next bucket
+ */
 export default class Aggregator {
 
-    constructor(size, processor, observer) {
-        this._generator = new Generator(size);
-        this._processor = processor;
-        this._bucket = null;
-        this._observer = observer;
-    }
-
-    /**
-     * Gets the current bucket or returns a new one.
-     *
-     * If a new bucket is generated the result of the old bucket is emitted
-     * automatically.
-     */
-    bucket(d) {
-        const thisBucketIndex = this._generator.bucketIndex(d);
-        const currentBucketIndex = this._bucket ?
-            this._bucket.index().asString() : "";
-        if (thisBucketIndex !== currentBucketIndex) {
-            if (this._bucket) {
-                this._bucket.aggregate(this._processor, event => {
-                    if (this._observer) {
-                        this._observer(this._bucket.index(), event);
-                    }
-                });
-            }
-            this._bucket = this._generator.bucket(d);
+    constructor(options, observer) {
```

Comment:
I don't think docs got updated for new constructor signature, looks like options is a object now, but docs have discrete arguments.